pkgs/top-level/all-packages.nix

```diff
@@ -6215,6 +6217,10 @@ with pkgs;
   rust = callPackage ../development/compilers/rust { };
   inherit (rust) cargo rustc;
 
+  mkRustCrate = callPackage ../build-support/rust/rust-utils.nix { };
```

Comment:
I usually try to avoid those discussion about naming things. Would like to stay with the name `mkRustCrate` to differentiate from `buildRustPackage`? Or would it not be better to name it `buildRustCrate` and become consistent with the other `builders`: `buildGoPackage`, `buildRubyGem`, `buildPythonPackage`, `buildPerlPackage`, `buildNodePackage`?

Reply:
I like `buildRustCrate` better, but don't have a strong opinion.

Reply:
This solution only builds rust crates? There are other artefacts possible such as shared objects. by using `buildRustPackage` you gave a generic enough name to express intent.

Reply:
Ultimately, this solution should build anything `cargo build` can build. It already does, for many crates on crates.io (and my own projects).

Reply:
yes indeed it does, I use your solution on my projects too. Though it's customized. I'll probably be removing that code in favour of this code. I renamed it to `build-rust-package` or to that effect and added an attribute `type = "crate"`. I have a few other build types already. (see https://github.com/fractalide/fractalide/blob/master/modules/rs/crates/default.nix#L62-L72 for more details)

Reply:
Ok, changed the name to `buildRustCrate`. Nix's documentation is "light" enough that being at least consistent with other languages already provide some documentation. I'll add more documentation for this tool now, though.